bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/CPU.java

```diff
@@ -2142,6 +2303,11 @@ private static void execTypeConversionOpcodes(WorkerExecutionContext ctx, Worker
                 j = operands[1];
                 sf.intRegs[j] = sf.doubleRegs[i] != 0.0 ? 1 : 0;
                 break;
+            case InstructionCodes.F2D:
+                i = operands[0];
+                j = operands[1];
+                sf.refRegs[j] = new BDecimal(new BFloat(sf.doubleRegs[i]).decimalValue());
```

Comment:
Same thing here, shall we directly create the BigDecimal here rather than wrapping it with BFloat?

Reply:
Addressed with commit ce04700.